numba/core/types/scalars.py

```diff
@@ -2,18 +2,34 @@
 
 import numpy as np
 
-from .abstract import Dummy, Hashable, Literal, Number, Type
+from .abstract import Dummy, Hashable, Literal, Number, Type, Bounded
 from functools import total_ordering
 from numba.core import utils
 from numba.core.typeconv import Conversion
 from numba.np import npdatetime_helpers
 
 
-class Boolean(Hashable):
+class Boolean(Hashable, Bounded):    
+    def __init__(self, name):
+        super(Boolean, self).__init__(name)
+        self.bitwidth = 1
```

Comment:
i am worry about adding this as a type property. `bitwidth` is an implementation detail. Numba represents it has an 1-bit value and an 8-bits value depending on how it's used.